src/Type/MixedType.php

```diff
@@ -526,6 +528,32 @@ public function toFloat(): Type
 
 	public function toString(): Type
 	{
+		if ($this->subtractedType !== null) {
+			$castsToEmptyString = new UnionType([
```

Comment:
In the same way there `AllowedArrayKeysTypes::getType` I dunno if it would be interesting to introduce a class/static methods for
- castsToZero
- castsToEmptyString
- castsToZeroString
- ...

Reply:
do you see a use-case where we need the same implementation somewhere else?

Reply:
No indeed, there is all the following union which is part similar but always different so far

NullType::getSmallerOrEqualType
```
		return new UnionType([
			new NullType(),
			new ConstantBooleanType(false),
			new ConstantIntegerType(0),
			new ConstantFloatType(0.0),
			new ConstantStringType(''),
			new ConstantArrayType([], []),
		]);
```

NullType::getGreaterType
```
return new MixedType(false, new UnionType([
			new NullType(),
			new ConstantBooleanType(false),
			new ConstantIntegerType(0),
			new ConstantFloatType(0.0),
			new ConstantStringType(''),
			new ConstantArrayType([], []),
		]));
```

```
$castsToZero = new UnionType([
			new NullType(),
			new ConstantBooleanType(false),
			new ConstantIntegerType(0),
			new ConstantArrayType([], []),
			new StringType(),
			new FloatType(), // every 0.x float casts to int(0)
		]);
```

```
$castsToEmptyString = new UnionType([
				new NullType(),
				new ConstantBooleanType(false),
				new ConstantStringType(''),
			]);
```

```
$castsToZeroString = new UnionType([
					new ConstantFloatType(0.0),
					new ConstantStringType('0'),
					new ConstantIntegerType(0),
				]);
```

StaticTypeFactory::falsey
```
$falsey = new UnionType([
				new NullType(),
				new ConstantBooleanType(false),
				new ConstantIntegerType(0),
				new ConstantFloatType(0.0),
				new ConstantStringType(''),
				new ConstantStringType('0'),
				new ConstantArrayType([], []),
			]);
```